samples/Azure.Analytics.Purview.Account/Generated/PurviewAccountResourceSetRules.cs

```diff
@@ -39,7 +39,7 @@ protected PurviewAccountResourceSetRules()
         /// <param name="tokenCredential"> The token credential to copy. </param>
         /// <param name="endpoint"> The account endpoint of your Purview account. Example: https://{accountName}.purview.azure.com/account/. </param>
         /// <param name="apiVersion"> Api Version. </param>
-        /// <exception cref="ArgumentNullException"> <paramref name="clientDiagnostics"/>, <paramref name="pipeline"/>, <paramref name="endpoint"/>, or <paramref name="apiVersion"/> is null. </exception>
+        /// <exception cref="ArgumentNullException"> <paramref name="clientDiagnostics"/>, <paramref name="pipeline"/>, <paramref name="endpoint"/> or <paramref name="apiVersion"/> is null. </exception>
         internal PurviewAccountResourceSetRules(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, TokenCredential tokenCredential, Uri endpoint, string apiVersion = "2019-11-01-preview")
         {
             Argument.AssertNotNull(clientDiagnostics, nameof(clientDiagnostics));
```

Comment:
(3 lines below -- re: apiVersion)
I had thought we didn't validate apiVersion in DPG clients because it would never be a string.  But it looks like it is a string here and we are validating it.  Could you help me understand this?

Reply:
It is a subclient, which gets its version from the parent.

Reply:
I actually wonder if we should generate any argument checks here at all. This constructor is not meant to be called from anywhere expect factory method of the parent client, which should guarantee that all parameters are correct.

Reply:
There's a chance at some point in the future ClientDiagnostics migrates to Core (this would take work, so I'm not saying it'll happen any time soon).  If that happened and this constructor became public, would we know to add back validation of arguments?  If so, I'm fine to take them out here.

Reply:
Sorry, but I'm not sure why would this constructor become public. Subclients are non-instantiable by design: https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-subclients

Reply:
Good point.